python/tank/platform/engine.py

```diff
@@ -452,6 +468,53 @@ def destroy_engine(self):
         Implemented by deriving classes.
         """
         pass
+
+    def change_context(self, new_context):
+        """
+        Called when the engine is being asked to change contexts. This
+        will only be allowed if the engine and all associated apps allow
+        a context change. In most cases this will also be implemented on
+        deriving classes to handle some DCC-specific operations, such as
+        menu reconstruction based on the new context. In those cases, this
+        parent change_context method should be called from the overriding
+        method first via super().
+        """
+        # Make sure we're allowed to change context at the engine level.
+        if not self.context_change_allowed:
+            raise TankContextChangeNotAllowedError()
+
+        # Check to see if all of our apps are capable of accepting
+        # a context change. If one of them is not, then we remove it
+        # from the persistent app pool, which will force it to be
+        # rebuilt when apps are loaded later on.
+        non_compliant_app_names = []
```

Comment:
oh! so here it looks like it doesn't require all apps to be context-switch compliant? Is it just the comment earlier in the code that got me confused? That's great? (maybe comment earlier on can be made clearer?). This is awesome!